frontend/src/components/pages/rp-connect/secrets/Secrets.List.tsx

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import {PageComponent, PageInitHelper} from '../../Page';
+import {observer} from 'mobx-react';
+import {appGlobal} from '../../../../state/appGlobal';
+import {rpcnSecretManagerApi} from '../../../../state/backendApi';
+import {Features} from '../../../../state/supportedFeatures';
+import {Box, Button, ButtonGroup, Code, ConfirmItemDeleteModal, createStandaloneToast, DataTable, Flex, Image, SearchField, Text} from '@redpanda-data/ui';
+import Section from '../../../misc/Section';
+import PageContent from '../../../misc/PageContent';
+import {uiSettings} from '../../../../state/ui';
+import {Link} from 'react-router-dom';
+import {PencilIcon, TrashIcon} from '@heroicons/react/outline';
+import EmptyConnectors from '../../../../assets/redpanda/EmptyConnectors.svg';
+import {DeleteSecretRequest, Secret} from '../../../../protogen/redpanda/api/dataplane/v1alpha2/secret_pb';
+
+const {ToastContainer, toast} = createStandaloneToast();
+
+const CreateSecretButton = () => {
+    return (<Box style={{display: 'flex', marginBottom: '.5em'}}>
+        <Link to={'/rp-connect/secret/create'}><Button variant="outline" colorScheme="brand">Create Secret</Button></Link>
+    </Box>)
+}
+
+const EmptyPlaceholder = () => {
+    return <Flex alignItems="center" justifyContent="center" flexDirection="column" gap="4" mb="4">
+        <Image src={EmptyConnectors}/>
+        <Box>You have no Redpanda Connect secrets.</Box>
+        <CreateSecretButton/>
+    </Flex>
+};
+
+@observer
+class RpConnectSecretsList extends PageComponent {
+
+    initPage(p: PageInitHelper) {
+        p.addBreadcrumb('Redpanda Connect Secret Manager', '/rp-connect/secrets');
+        this.refreshData(true);
+        appGlobal.onRefresh = () => this.refreshData(true);
+    }
+
+    refreshData(force: boolean) {
+        if (!Features.pipelinesApi) return;
+
+        rpcnSecretManagerApi.refreshSecrets(force)
+            .catch((err) => {
+                if (String(err).includes('404')) {
+                    // Hacky special handling for OSS version, it is possible for the /endpoints request to not complete in time for this to render
+                    // so in this case there would be an error shown because we were too fast (with rendering, or the req was too slow)
+                    // We don't want to show an error in that case
+                    return;
+                }
```

Comment:
I think maybe it sounds worse than it is, but thanks for explaining 😬

Reply:
I copy and past this handle.